packages/scaffold-config/src/frameworks.ts

```diff
@@ -0,0 +1,579 @@
+    packages: [CYPRESS_VUE_3],
+    defaultPackagePath: null,
+    glob: '*.vue',
+    detectors: [
+      {
+        dependency: 'vue',
+        version: '^3.0.0',
+      },
+    ],
+    category: FRONTEND_FRAMEWORK_CATEGORIES[1],
+    codeGenFramework: CODE_GEN_FRAMEWORKS[1],
+    storybookDep: STORYBOOK_VUE,
+    config: {
+      js: (bundler: Bundler) => {
+        if (isWebpack(bundler)) {
+          return dedent`
+          const { devServer } = require('@cypress/webpack-dev-server')
+          // NOTE: ensure you are requiring your webpack config from the
+          // correct location.
+          const webpackConfig = require('./webpack.config.js')
+
+          module.exports = {
+            component: {
+              devServer,
+              devServerConfig: {
+                webpackConfig
+              }
+            }
+          }`
+        }
+
+        if (bundler === 'vite') {
+          return dedent`
+          const { devServer } = require('@cypress/vite-dev-server')
+
+          module.exports = {
+            component: {
+              devServer,
+              devServerConfig: {
+                // optionally provide your Vite config overrides.
+              }
+            }
+          }`
+        }
+
+        throw Error(`No config defined for ${bundler}`)
+      },
+
+      ts: (bundler: Bundler) => {
+        if (isWebpack(bundler)) {
+          return dedent`
+          import { defineConfig } from 'cypress'
+          import { devServer } from '@cypress/webpack-dev-server'
+          import webpackConfig from '@vue/cli-service/webpack.config'
+
+          export default defineConfig({
+            component: {
+              devServer,
+              devServerConfig: {
+                webpackConfig
+              }
+            }
+          })`
+        }
+
+        if (bundler === 'vite') {
+          return dedent`
+          import { defineConfig } from 'cypress'
+          import { devServer } from '@cypress/vite-dev-server'
+
+          export default defineConfig({
+            component: {
+              devServer,
+                // optionally provide your Vite config overrides.
+              devServerConfig: {}
+            }
+          })`
+        }
+
+        throw Error(`No config defined for ${bundler}`)
+      },
+    },
+  },
+  {
+    type: 'nextjs',
+    name: 'Next.js',
+    family: 'template',
+    supportedBundlers: [BUNDLER_WEBPACK_4],
+    packages: [
+      CYPRESS_REACT_LATEST,
+      CYPRESS_WEBPACK,
+      WEBPACK_DEV_SERVER_4,
+      HTML_WEBPACK_PLUGIN_4,
+    ],
+    defaultPackagePath: '@cypress/react/plugins/next',
+    glob: '*.{js,jsx,tsx}',
+    category: FRONTEND_FRAMEWORK_CATEGORIES[0],
+    detectors: [
+      {
+        dependency: 'next',
+        version: '>=10.0.0',
+      },
+    ],
+    codeGenFramework: CODE_GEN_FRAMEWORKS[0],
+    storybookDep: STORYBOOK_REACT,
+    config: {
+      js: () => ``,
+      ts: () => ``,
```

Comment:
Next.js 11/12 is not working properly with Cypress CT yet. Once we find out why, we can add this.